capstone/src/main/webapp/script.js

```diff
@@ -12,44 +12,219 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/**
- * Call the perspective API
- */
-async function callPerspective(text, lang) {
-  const response = await fetch('/call_perspective', {
-      method: 'POST',
-      headers: {'Content-Type': 'application/json',},
-      body: JSON.stringify({text: text, lang: lang})});
-  const toxicityData = await response.json();
-  displayPerspectiveOutput(toxicityData);
-}
-
-/**
- *  Display toxicity output on the webpage
- */
-function displayPerspectiveOutput(toxicityData) {
-  const outputElement = document.getElementById('perspective-output-container');
-  if (!outputElement) {
+const ATTRIBUTES_BY_LANGUAGE = {
+  'en': ['TOXICITY', 'SEVERE_TOXICITY', 'TOXICITY_FAST', 'IDENTITY_ATTACK', 'INSULT', 'PROFANITY', 'THREAT', 'SEXUALLY_EXPLICIT', 'FLIRTATION'],
+  'es': ['TOXICITY', 'SEVERE_TOXICITY', 'IDENTITY_ATTACK_EXPERIMENTAL', 'INSULT_EXPERIMENTAL', 'PROFANITY_EXPERIMENTAL', 'THREAT_EXPERIMENTAL'],
+  'fr': ['TOXICITY', 'SEVERE_TOXICITY', 'IDENTITY_ATTACK_EXPERIMENTAL', 'INSULT_EXPERIMENTAL', 'PROFANITY_EXPERIMENTAL', 'THREAT_EXPERIMENTAL'],
+  'de': ['TOXICITY', 'SEVERE_TOXICITY', 'IDENTITY_ATTACK', 'INSULT', 'PROFANITY', 'THREAT'],
+  'it': ['TOXICITY', 'SEVERE_TOXICITY', 'IDENTITY_ATTACK', 'INSULT', 'PROFANITY', 'THREAT'],
+  'pt': ['TOXICITY', 'SEVERE_TOXICITY', 'IDENTITY_ATTACK', 'INSULT', 'PROFANITY', 'THREAT']
+};
+
+/** Collects the user's input and produces Perspective analysis */
+async function handleInput() {
+  // Get the submitted text and language
+  const textElement = document.getElementById('textForAnalysis');
+  if (!textElement) {
     return;
   }
-  outputElement.innerHTML = '';
+  const langElement = document.getElementById('languageForAnalysis');
+  if (!langElement) {
+    return;
+  }
+
+  // Get the selected attributes
+  const attributes = document.getElementById("available-attributes").getElementsByTagName('input');
+  const requestedAttributes = []
+  for (let attribute of attributes) {
+    if (attribute.checked) {
+      requestedAttributes.push(attribute.value);	
+    }	
+  }
+
```

Comment:
For modularity, I think it might be useful to slice this function into two parts here; one that gets the input, language, and attributes, and another that calls perspective/handles output. We might have an easier time later on when different pages have different ways of gathering user input.

Reply:
Good idea - I split it into `gatherInput()` and `handleInput()`